cdap-ui/app/cdap/components/Experiments/store/ActionCreator.js

```diff
@@ -225,6 +252,47 @@ function getModelStatus(experimentId, modelId) {
           modelStatus
         }
       });
+    }, () => {
+      experimentDetailsStore.dispatch({
+        type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
+        payload: {
+          modelId,
+          modelStatus: 'error'
+        }
+      });
+    });
+}
+
+function getModelStatus(experimentId, modelId) {
+  experimentDetailsStore.dispatch({
+    type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
+    payload: {
+      modelId,
+      modelStatus: 'loading'
+    }
+  });
+  myExperimentsApi
+    .getModelStatus({
+      namespace: getCurrentNamespace(),
+      experimentId,
+      modelId
+    })
+    .subscribe(modelStatus => {
+      experimentDetailsStore.dispatch({
+        type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
+        payload: {
+          modelId,
+          modelStatus
+        }
+      });
+    }, () => {
+      experimentDetailsStore.dispatch({
+        type: EXPERIMENTDETAILACTIONS.SET_MODEL_STATUS,
+        payload: {
+          modelId,
+          modelStatus: 'error'
```

Comment:
This is not the "model status" from the backend right? If we are showing this in UI will we cause inconsistency with what is the state in backend? Right now there is a specific cycle in backend for setting model status. https://wiki.cask.co/display/CE/Athena+-+Model+Management+and+Distribution+Service I don't believe us setting the status as `error` is correct here.